system/hostname/hostname_main.c

```diff
@@ -42,17 +42,64 @@
 int main(int argc, FAR char *argv[])
 {
   int ret;
+  FILE *f;
+  int maxarg = 2;
   char hostname[HOST_NAME_MAX + 1];
 
   if (argc > 1)
     {
-      if (argc > 2)
+      if (strcmp(argv[1], "-F") == 0)
+        {
+          if (argc < 3)
+            {
+              fprintf(stderr, "ERROR: Option requires an argument -- 'F'\n");
+              return EXIT_FAILURE;
+            }
+
+          f = fopen(argv[2], "r");
+          if (f == NULL)
+            {
+              fprintf(stderr, "ERROR: Failed to open '%s': %s\n", argv[2],
+                      strerror(errno));
+              return EXIT_FAILURE;
+            }
+
+          if (fgets(hostname, sizeof(hostname), f) == NULL && (errno != 0))
+            {
+              fclose(f);
```

Comment:
`fclose` may modify `errno`, `fprintf` may print the value different from one that met `(errno != 0)` to true in the `if`.